### PR TITLE
Merging events domain and name

### PR DESCRIPTION
**Pull request by patrickhousley**
Fixes [#2994](https://github.com/open-telemetry/opentelemetry-specification/issues/2994)

## Changes

Merging the `domain` and `name` fields for events and modifying language to refer to the first part of the name as `namespace`. This more closely aligns the semantic conventions of `events` with `metrics` as we have discussed during the events working group meeting.

The spec update pull request is [here](https://github.com/open-telemetry/opentelemetry-specification/pull/3749).

Note: if the PR is touching an area that is not listed in the [existing areas](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/README.md), or the area does not have sufficient [domain experts coverage](https://github.com/open-telemetry/semantic-conventions/blob/main/.github/CODEOWNERS), the PR might be tagged as [experts needed](https://github.com/open-telemetry/semantic-conventions/labels/experts%20needed) and move slowly until experts are identified.

## Merge requirement checklist

* [ ] [CONTRIBUTING.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md) guidelines followed.
* [ ] [CHANGELOG.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CHANGELOG.md) updated for non-trivial changes.
* [ ] [schema-next.yaml](https://github.com/open-telemetry/semantic-conventions/blob/main/schema-next.yaml) updated with changes to existing conventions.

### CODE REVIEW

File: CHANGELOG.md
Comment (MSNev) [inline CHANGELOG.md:41]: change rather than chance :-)

File: docs/general/events.md
Comment (breedx-splk) [inline docs/general/events.md:22]:
```suggestion
`event.name` consisting of a dot-separated `namespace` and `name` that
```

**Review by breedx-splk — APPROVED** [review]
Looks good to me, and matches what we talked about in the sig meeting. Do the protobuf definitions also need to change accordingly? Can we link to that work or create a follow-up issue? Thanks! 🙏🏻

**Comment by patrickhousley** [comment]
> Looks good to me, and matches what we talked about in the sig meeting. Do the protobuf definitions also need to change accordingly? Can we link to that work or create a follow-up issue? Thanks! 🙏🏻

I don't see events in the protobuf definitions repo. I may be blind so could you drop me a link? If it's not there, we should get a ticket created to make the protobuf definition.

File: docs/general/events.md
Comment (tigrannajaryan) [inline docs/general/events.md:23]:
```suggestion
uniquely identifies the event. Events may also have other LogRecord attributes.
```

File: docs/general/events.md
Comment (tigrannajaryan) [inline docs/general/events.md:34]:
```suggestion
Events with same `event.name` are structurally similar to one another.
```

File: docs/general/events.md
Comment (tigrannajaryan) [inline docs/general/events.md:49]:
```suggestion
```

File: docs/general/events.md
Comment (tigrannajaryan) [inline docs/general/events.md:57]: I think we should get rid of this and instead say that event namespaces are shared with all other semconv namespaces.

**Comment by tigrannajaryan** [comment]
The PR needs a description that explains why the change is made, the motivation and reasoning.

It links to issue #2994 which does not exist in this repo.

**Review by tigrannajaryan — CHANGES_REQUESTED** [review]
I am temporarily blocking the PR to give time to @scheler as they requested: https://github.com/open-telemetry/opentelemetry-specification/issues/2994#issuecomment-1785633470

@scheler please comment on how much time you think you need.

The work on this PR can continue, it is a temporary block to avoid premature merging.

File: model/logs/events.yaml
Comment (AlexanderWert) [inline model/logs/events.yaml:15]: Sorry for being late on/having missed the discussion (that started in https://github.com/open-telemetry/opentelemetry-specification/issues/2994).

But, with that change I'm wondering how users will be able to filter / query data by the `name` part only?

Let's say I want to see all `exception` events. Now, it's quite simple with `event.name = exception`.
With that change all of this will change to `device.exception`, `browser.exception`, `mycustom.exception`, etc. So how to get ALL types of exceptions?

Also, what is the implication for the [exceptions semantic conventions on span events](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/exceptions/exceptions-spans.md#recording-an-exception)?:

> An exception SHOULD be recorded as an Event on the span during which it occurred. The name of the event MUST be "exception".

Would that be a plain `exception`, or any prefix expected?

Comment (tigrannajaryan) [reply model/logs/events.yaml:15]: Good catch, thanks @AlexanderWert 

The way https://github.com/open-telemetry/semantic-conventions/blob/main/docs/exceptions/exceptions-spans.md#recording-an-exception is defined it appears `exception.` is the namespace, not the event name (e.g. we have `exception.message` as the attribute).

I think that aligns nicely with this PR.

We may need to discuss if `exception.` is the the right root name for the namespace or we want to change it.

Comment (AlexanderWert) [reply model/logs/events.yaml:15]: @tigrannajaryan Not sure I understand correctly.

But in https://github.com/open-telemetry/semantic-conventions/blob/main/docs/exceptions/exceptions-spans.md#recording-an-exception it's written:

> The name of the event MUST be "exception".

So, I'm not referring to the namespace `exception.*`, but on the fact that currently the `event.name` is being set to the value `exception` (and `event.domain` to the domain in which the exception occurred, I guess?). And I'm wondering what the value for `event.name` would be for exceptions with this change here.

File: docs/general/events.md
Comment (patrickhousley) [reply docs/general/events.md]: I removed this table and blurb for now and I think the following section pointing to the attribute naming is enough for now. I do think we will probably have a table in here at some point for well known namespace prefixes once we come to an agreement in the WG.

File: docs/general/events.md
Comment (jack-berg) [inline docs/general/events.md:25]: Rather than define the namespace concept in this document, I think we should link out to the [attribute naming](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/common/attribute-naming.md) document and say something to the effect of: "Event names are subject to the same rules as [attribute names](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/common/attribute-naming.md). Notably, event names are namespaced to avoid collisions."

File: model/logs/events.yaml
Comment (jack-berg) [reply model/logs/events.yaml:15]: The event name uniquely defines the type / class of thing that happened. The namespace portion of the event name is meant to avoid collisions, since we may eventually define multiple events called "click" with different semantic meanings. 

For the span event exception API, we'll need to define how the existing semantics of "a span event with name 'exception'", map to the event API and the event semantic conventions. Essentially, when recordException is called, we'll need to define what to assign to the event.name field, and which fields are included in the payload. The payload of record exception seems self explanatory. The event.name field should probably include "exception" as the suffix, but we need to choose an appropriate namespace. I don't think we want to have different namespaces for different environment (i.e. we don't want device.exception, browser.exception, etc). Instead, we should define a environment / runtime agnostic way of indicting that an exception occurred. Maybe the name is something like `runtime.exception` or something similar?

File: docs/general/events.md
Comment (patrickhousley) [reply docs/general/events.md]: @jack-berg take another look pls. I also updated the general info to remove the blurbs around defining namespace concept.

File: docs/general/events.md
Comment (jack-berg) [inline docs/general/events.md:36]: Can we find a different example than `browser.exception`? I don't envision having different exception events for different runtimes. How about using the `device.app.lifecycle` event name defined here? https://github.com/open-telemetry/semantic-conventions/blob/main/docs/mobile/events.md

(Side note, we'll want to update the device app.lifecycle event and any others like it that specify a event.domain.

File: docs/general/events.md
Comment (jack-berg) [reply docs/general/events.md]: 👍

File: model/logs/events.yaml
Comment (scheler) [reply model/logs/events.yaml:15]: One possibility that a separate `event.domain` enabled was to route all events from different domains to different destinations, for eg., different indices in data stores. With the domain merged as namespace into the name, it's still doable but with some complexity. If browser exceptions will not have browser as prefix, I guess we will have to use `Resource` for the purpose.

Comment (tigrannajaryan) [reply model/logs/events.yaml:15]: >If browser exceptions will not have browser as prefix, I guess we will have to use Resource for the purpose.

Yes. One kink here is that none of [`browser.`](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/resource/browser.md) Resource attributes are "Required" so you can't 100% rely on the presence. I don't know why isn't at least one browser attribute required, maybe it should.

Comment (scheler) [reply model/logs/events.yaml:15]: I submitted https://github.com/open-telemetry/semantic-conventions/issues/511 to make at least one browser attribute required.

**Review by AlexanderWert — APPROVED** [review]

File: docs/general/events.md
Comment (AlexanderWert) [reply docs/general/events.md:36]: +1

Comment (AlexanderWert) [reply docs/general/events.md:36]: How about `device.app.lifecycle` or `mobile.app.lifecycle` ?

Comment (patrickhousley) [reply docs/general/events.md:36]: I think we need to decide on what our prefixes should look like before I change this. Do we need to use `browser` and `mobile` or `device`? What other prefixes would be acceptable and how do we tell consumers about them?

Comment (jack-berg) [reply docs/general/events.md:36]: This PR is all about the merging of `event.domain` and `event.name`. The natural way to merge these is to concatenate `event.domain + event.name`. So let's update the [mobile lifecycle events](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/mobile/events.md#lifecycle-instrumentation) to have event.names of `device.app.lifecycle`.

> What other prefixes would be acceptable and how do we tell consumers about them?

The [attribute naming](https://github.com/open-telemetry/opentelemetry-specification/tree/v1.26.0/specification/common/attribute-naming.md) document you've linked to defines the rules for `event.name`.

File: model/logs/events.yaml
Comment (MSNev) [reply model/logs/events.yaml:15]: Pretty much what most have said here, but in practice I would not "expect" anyone to 

> Let's say I want to see all exception events.

Unless they are the "same" common thing (which is probably unlikely for this example), but if they are the same then it should be the same `event.name`. 

Simplistically, a defined `event.name` (fully qualified definition of an `event`) should be unique and it should (generally) not make any sense to group directly with other `event.name`s. The exception to this would be if specific events are explicitly related or someone is trying to group them based on the same session.id etc. At which point searching / linking based on (only) the name would probably not be possible anyway.

**Review by MSNev — APPROVED** [review]

File: docs/general/events.md
Comment (jack-berg) [inline docs/general/events.md:36]:
```suggestion
| `event.name` | string | Identifies the class / type of event. | `browser.mouse.click`; `browser.exception` | Required |
```
The phrasing of "Uniquely identifies" reads like this could be some sort of UUID type identifier which is unique for every distinct record.

Comment (patrickhousley) [reply docs/general/events.md:36]: Should we do that in this PR or a separate PR?

Comment (jack-berg) [reply docs/general/events.md]: I think this PR. The [mobile lifecylce events](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/mobile/events.md#lifecycle-instrumentation) is the only place where an `event.domain` / `event.name` is specified, so should be small in scope.

Comment (patrickhousley) [reply docs/general/events.md]: Done.

Comment (jack-berg) [reply docs/general/events.md]: Marking this comment as "unresolved" because `browser.exception` is still used as an example. Can we switch to `device.app.lifecycle` instead?

File: docs/general/events.md
Comment (patrickhousley) [reply docs/general/events.md:38]: Should we not point to a specific version of the specification?

Comment (tigrannajaryan) [reply docs/general/events.md:38]: No, I don't think we should. Spec is versioned all together, all of its documents. If the linked document is updated it should apply to this doc too.

File: docs/general/events.md
Comment (patrickhousley) [reply docs/general/events.md]: Should be good now.

File: docs/general/events.md
Comment (patrickhousley) [reply docs/general/events.md:38]: The specs are linked to in a lot of other docs in this same manner. Should we update all of them and is that necessarily an ask or work that is outside the scope of this PR?

Comment (tigrannajaryan) [reply docs/general/events.md:38]: No need to update other links in this PR.

I wonder if any reviewer has any arguments in favour of linking to a specific version? I thought it is a mistake but since it is in many places perhaps it is deliberate and in that case I wonder what the reason is.

Comment (scheler) [reply docs/general/events.md:38]: Linking to a specific version will be confusing unless the version is part of the link text and is specifically called out.

**Review by jack-berg — APPROVED** [review]

**Review by scheler — APPROVED** [review]

**Comment by breedx-splk** [comment]
We're at 5 approvals now, so I think we're getting close. Looks like there's a table issue right now @patrickhousley 
```
File /spec/general/events.md contains a table that would be reformatted.
make: *** [Makefile:98: table-check] Error 1
```

**Comment by patrickhousley** [comment]
> We're at 5 approvals now, so I think we're getting close. Looks like there's a table issue right now @patrickhousley
> 
> ```
> File /spec/general/events.md contains a table that would be reformatted.
> make: *** [Makefile:98: table-check] Error 1
> ```

I run `make fix` locally but it's like it's reverting changes to the files. I am not sure what the error is or what the fix is doing.

File: docs/mobile/events.md
Comment (AlexanderWert) [inline docs/mobile/events.md:27]: The section between `<!-- semconv ios.lifecycle.events -->` and `<!-- endsemconv -->` is auto-generated.

Please do the actual changes in the corresponding yaml file, i.e. following places:

- https://github.com/open-telemetry/semantic-conventions/blob/126059d99dc29fd9a59893091f56500d4b9674b7/model/logs/mobile-events.yaml#L5
- https://github.com/open-telemetry/semantic-conventions/blob/126059d99dc29fd9a59893091f56500d4b9674b7/model/logs/mobile-events.yaml#L44

And then, please run `make table-generation` to re-generate that table. This will also fix the current error:

> File /spec/general/events.md contains a table that would be reformatted.
make: *** [Makefile:98: table-check] Error 1

**Review by AlexanderWert — CHANGES_REQUESTED** [review]

File: docs/mobile/events.md
Comment (patrickhousley) [reply docs/mobile/events.md:27]: Ahhh ok thanks @AlexanderWert .

**Comment by patrickhousley** [comment]
@breedx-splk I believe we are good to go now.

File: docs/general/events.md
Comment (tigrannajaryan) [inline docs/general/events.md:38]: I think the link needs to go to main branch.

**Review by tigrannajaryan — CHANGES_REQUESTED** [review]

File: docs/general/events.md
Comment (patrickhousley) [reply docs/general/events.md]: https://github.com/open-telemetry/semantic-conventions/issues/549 opened to update the other links in the repo.

File: docs/general/events.md
Comment (patrickhousley) [reply docs/general/events.md]: https://github.com/open-telemetry/semantic-conventions/issues/549 opened to update the other links in the repo.

File: model/logs/events.yaml
Comment (patrickhousley) [reply model/logs/events.yaml]: Is there any action items from this discussion that need to be accounted for before merging? If so, can those be turned into issues please?

File: model/logs/events.yaml
Comment (joaopgrassi) [inline model/logs/events.yaml:14]: This should not point to main. This repo targets a specific spec release, which is mentioned in here: https://github.com/open-telemetry/semantic-conventions#opentelemetry-semantic-conventions

> This repository is currently using [this specification version](https://github.com/open-telemetry/opentelemetry-specification/tree/v1.26.0).

And in CONTRIBUTION there's a guide for maintainers in how to update the spec version in the entire repo: https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md#updating-the-referenced-specification-version

**Review by joaopgrassi — CHANGES_REQUESTED** [review]

**Review by AlexanderWert — APPROVED** [review]

File: model/logs/events.yaml
Comment (tigrannajaryan) [reply model/logs/events.yaml:14]: Thanks @joaopgrassi for pointing this out, this is the part I was looking for.

File: docs/general/events.md
Comment (tigrannajaryan) [reply docs/general/events.md]: I was wrong, see https://github.com/open-telemetry/semantic-conventions/pull/473#discussion_r1401926563

**Review by tigrannajaryan — APPROVED** [review]

**Review by joaopgrassi — CHANGES_REQUESTED** [review]
Please change the links to the spec repo to the correct tag.

File: CHANGELOG.md
Comment (joaopgrassi) [inline CHANGELOG.md:21]: Nit: This could be combined into a single changelog entry. 

> - BREAKING: Remove `event.domain` from log event attributes and adapt `event.name` definition to include `namespace`

**Review by joaopgrassi — COMMENTED** [review]
Looks good, but the PR does more things then it says in the description. It also adds the `device` prefix, which I couldn't find any decision or discussion about it in the comments (maybe I missed it?)

I saw this issue https://github.com/open-telemetry/semantic-conventions/issues/548 which opens the discussion but also no conclusion. Since this has several approvals, is everyone OK with this change as well? Is everyone aware of this?

**Comment by AlexanderWert** [comment]
> It also adds the device prefix, which I couldn't find any decision or discussion about it in the comments (maybe I missed it?)

Actually it does not __add__ the device prefix, but for mobile merges the `event.domain`(which has the value `device`) into the `event.name` as a prefix. So, actually applies the general proposal to the only relevant, concrete use case so far.

**Review by joaopgrassi — APPROVED** [review]

**Pull request merged**
